src/txn/txn_rollback_to_stable.c

```diff
@@ -110,6 +110,10 @@ __txn_abort_newer_update(
             upd->durable_ts = upd->start_ts = WT_TS_NONE;
         }
     }
+
+    /* Reset the history store flag for the stable update. */
+    if (first_upd)
+        first_upd->flags |= WT_UPDATE_HS;
```

Comment:
The function handles `first_upd == NULL`, but one of its callers checks before making the call. It seems that's inconsistent?

Reply:
`upd` is used to check for the NULL condition in the above `for` loop. Once the loop is finished there should update that satisfy the given timestamp or there are no more updates. So NULL check is required before resetting the flag.

Reply:
I'm talking about the function itself -- some (but not all) of the callers check if they're passing a NULL as the `first_upd` argument to the function.

Reply:
Fixed.